python/xorbits/_mars/dataframe/sort/psrs.py

```diff
@@ -660,15 +660,77 @@ def _execute_series_map(cls, ctx, op):
                 values = a.iloc[poses[i] : poses[i + 1]]
                 ctx[out.key, (i,)] = values
 
+    @classmethod
+    def _calc_pivots_poses(
+        cls, index, pivots, is_multi_index=False, level=None, sort_remaining=True
+    ):
+        if not is_multi_index:
+            return index.searchsorted(list(pivots), side="right")
+
+        if index.nlevels != pivots.nlevels:
+            raise TypeError(
+                f"Number of levels should be the same between index and pivots. But got nlevels of index is {index.nlevels}, and nlevels of pivots is {pivots.nlevels}"
```

Comment:
Better make error message more consise. Let ChatGPT do it for you :)

```suggestion
                f"Inconsistent levels: index has {index.nlevels} levels, while pivots has {pivots.nlevels} levels."
```

Reply:
done, thx